pkg/cloud/openstack/openstack.go

```diff
@@ -0,0 +1,48 @@
+package openstack
+
+import (
+	"bytes"
+	"embed"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	k8sYaml "k8s.io/apimachinery/pkg/util/yaml"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:embed assets/*
+var f embed.FS
+
+func GetResources() []client.Object {
+	resources := []struct {
+		obj   client.Object
+		asset string
+	}{
+		{&v1.ConfigMap{}, "assets/config.yaml"},
+		{&appsv1.Deployment{}, "assets/deployment.yaml"},
+		{&v1.ServiceAccount{}, "assets/rbac/serviceaccount.yaml"},
+		{&rbacv1.Role{}, "assets/rbac/role.yaml"},
+		{&rbacv1.RoleBinding{}, "assets/rbac/rolebinding.yaml"},
+		{&rbacv1.ClusterRoleBinding{}, "assets/rbac/clusterrolebinding.yaml"},
+	}
+	ret := make([]client.Object, 0, len(resources))
+
+	for i := 0; i < len(resources); i++ {
+		resource := &resources[i]
+		data, err := f.ReadFile(resource.asset)
+		if err != nil {
+			return nil
+		}
+
+		dec := k8sYaml.NewYAMLOrJSONDecoder(bytes.NewReader(data), 1000)
+		if err := dec.Decode(resource.obj); err != nil {
```

Comment:
On the topic of error handling...

I wonder if we should move static asset parsing to an init block and panic on failure. @Danil-Grigorev?

Reply:
I think we should properly pass cluster configuration in these methods and handle errors as usual. For now logging the error will be sufficient.

Reply:
I was just thinking that it's a 'static' failure. That is, the only resolution is to build a new image. Does continuing to run serve any purpose if it's never going to work? Presumably we would never ship anything in this state, but it would also make debugging the source of a CI failure simpler.

Reply:
IMO we should be panicking if we can't decode something that we have included in the binary by embedding. To me this signifies a programming error, hence a panic is suitable.

If the controller panics when it can't parse an embedded resource, this will be flagged up in tests and should prevent anyone checking in broken YAML. 

Additionally, a unit test that runs `GetResources` would be obviously broken if it panics and would flag any errors being added

Reply:
I added logging to show proper error messages